### PR TITLE
Add CRUD support for Images Variants

### LINKED ISSUE #1495: Add Support for Images Variants
### Current cloudflare-go version

0.87.0

### Description

#

### Use cases

https://github.com/cloudflare/terraform-provider-cloudflare

### Potential cloudflare-go usage

```go
It should work as normal crud so it can be used in the TF provider.
```

### References

_No response_

### PR DESCRIPTION
**Pull request by hwrdprkns**
## Description

Adds support for the [Images Variants](https://developers.cloudflare.com/api/operations/cloudflare-images-variants-list-variants) API.

Resolves #1495.

## Has your change been tested?

- Tested `ListImageVariantDetails` manually
- Tested `GetImageVariantDetails` manually
- Go Test

## Types of changes

What sort of change does your code introduce/modify?

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] This change is using publicly documented in [cloudflare/api-schemas](https://github.com/cloudflare/api-schemas) 
      and relies on stable APIs.

[1]: https://help.github.com/articles/closing-issues-using-keywords/

### CODE REVIEW

File: images_variants.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+		return ListImageVariantsResult{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return listImageVariantsResponse.Result, nil
+}
+
+// Fetch details for a single variant.
+//
+// API Reference: https://developers.cloudflare.com/api/operations/cloudflare-images-variants-variant-details
+func (api *API) GetImagesVariantDetails(ctx context.Context, rc *ResourceContainer, variantID string) (ImagesVariant, error) {
```
Comment (jacobbednarz) [inline images_variants.go]:
```suggestion
func (api *API) GetImagesVariant(ctx context.Context, rc *ResourceContainer, variantID string) (ImagesVariant, error) {
```

File: images_variants.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	Result ListImageVariantsResult `json:"result,omitempty"`
+	Response
+}
+
+type ListImageVariantsResult struct {
+	ImagesVariants map[string]ImagesVariant `json:"variants,omitempty"`
+}
+
+type CreateImagesVariantParams struct {
+	ImagesVariant
```
Comment (jacobbednarz) [inline images_variants.go]: this should be inlined to allow the structs to change independently

```
	ID                     string                `json:"id,omitempty"`
	NeverRequireSignedURLs *bool                 `json:"neverRequireSignedURLs,omitempty"`
	Options                ImagesVariantsOptions `json:"options,omitempty"`
```

**Review by jacobbednarz — CHANGES_REQUESTED** [review]
getting very close! just a couple of minor updates and we're good to merge this one.

**Comment by hwrdprkns** [comment]
@jacobbednarz Sounds good. Just made the updates - thanks for reviewing!

**Review by jacobbednarz — APPROVED** [review]

**Comment by hwrdprkns** [comment]
Thank you!

**Pull request merged**